test/extended/templates/helpers.go

```diff
@@ -58,3 +78,90 @@ func setUser(cli *exutil.CLI, user *userapi.User) {
 		cli.ChangeUser(user.Name)
 	}
 }
+
+func EnsureTSB(tsbOC *exutil.CLI) (osbclient.Client, *exec.Cmd) {
+	exists := true
+	if _, err := tsbOC.AdminKubeClient().Extensions().DaemonSets(tsbNS).Get("apiserver", metav1.GetOptions{}); err != nil {
+		if !kerrors.IsNotFound(err) {
+			o.Expect(err).NotTo(o.HaveOccurred())
+		}
+		exists = false
+	}
+
+	if !exists {
+		e2e.Logf("Installing TSB onto the cluster for testing")
+		_, _, err := tsbOC.AsAdmin().WithoutNamespace().Run("create", "namespace").Args(tsbNS).Outputs()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		configPath := exutil.FixturePath("..", "..", "examples", "templateservicebroker", "templateservicebroker-template.yaml")
+		stdout, _, err := tsbOC.WithoutNamespace().Run("process").Args("-f", configPath).Outputs()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = tsbOC.WithoutNamespace().AsAdmin().Run("create").Args("-f", "-").InputString(stdout).Execute()
+		// this is weird, but we have to ignore this error in case some of this already exists (race between tests and the like)
+		//o.Expect(err).NotTo(o.HaveOccurred())
+		err = WaitForDaemonSetStatus(tsbOC.AdminKubeClient(), &extensions.DaemonSet{ObjectMeta: metav1.ObjectMeta{Name: "apiserver", Namespace: tsbNS}})
+		o.Expect(err).NotTo(o.HaveOccurred())
+	}
+
+	// we're trying to test the TSB, not the service.  We're outside all the normal networks.  Run a portforward to a particular
```

Comment:
> Anyway yes it can wait, I just figured it would remove this port forwarding complexity at the same time, thus a net win.

Are you familiar with how the routers are built in our e2e envs?  I'm not, so I'd prefer a followup.